nixos/modules/misc/nixpkgs.nix

```
@@ -417,6 +434,11 @@ in
           '';
         }
       ];
+
+    nixpkgs.config.allowUnfreePredicate = lib.mkIf (
+      builtins.length config.nixpkgs.allowUnfreePackages > 0
+    ) (pkg: builtins.elem (lib.getName pkg) config.nixpkgs.allowUnfreePackages);
```

To make this more scalable:

```suggestion
    ) (pkg: allowUnfreePackagesAsSet?${lib.getName pkg});
```

And then in a `let` binding outside of the `pkg:` function, something like

```nix
lib.genAttrs (_name: null) config.nixpkgs.allowUnfreePackages
```

---

Is there a way to get a feel for the performance tradeoffs of these kind of changes? In procedural languages, it is quite often faster to use lists for these kind of tests - for small lists of around 8 or less items, than produce a hash and add everything to it.

This of course hinges on the assumption that this is only called for the actual unfree packages that are in my closure, which should be usually be exactly as many as are in this list.